File: src/Billion.java
class Billion {
    public static void main(String[] args) {
        for (int i = 0; i < 1000 * 1000 * 1000; i++)
        {
            double j = i / 9.0;
            if (i % (100 * 1000 * 1000) == 0)
            {
                System.out.println(j);
            }
        }
    }
}

File: src/Haskell/Billion.hs
billion = 1000 * 1000 * 1000
divisor = 100 * 1000 * 1000

loop :: Int -> IO ()
loop i
    | i `mod` divisor == 0 = do
    print j
    loop $ i + 1
    | i < billion = loop $ i + 1
    | otherwise = return ()
    where j = fromIntegral i / 9.0

main = loop 0
